src/System.ComponentModel.Annotations/tests/DisplayFormatAttributeTest.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.ComponentModel.DataAnnotations.Tests
+{
+    public class DisplayFormatAttributeTest
+    {
+        [Fact]
+        public void Ctor()
+        {
+            DisplayFormatAttribute attribute = new DisplayFormatAttribute();
+            Assert.True(attribute.ConvertEmptyStringToNull);
+            Assert.True(attribute.HtmlEncode);
+            Assert.False(attribute.ApplyFormatInEditMode);
+
+            Assert.Null(attribute.DataFormatString);
+            Assert.Null(attribute.NullDisplayText);
+            Assert.Null(attribute.NullDisplayTextResourceType);
+        }
+
+        [Theory]
+        [InlineData("{0:C}")]
+        [InlineData("{0:d}")]
+        public void DataFormatString_Get_Set(string input)
+        {
+            DisplayFormatAttribute attribute = new DisplayFormatAttribute();
+            attribute.DataFormatString = input;
+
+            Assert.Equal(input, attribute.DataFormatString);
+        }
+
+        [Fact]
+        public void ConvertEmptyStringToNull_Get_Set()
+        {
+            DisplayFormatAttribute attribute = new DisplayFormatAttribute();
+            attribute.ConvertEmptyStringToNull = false;
+
+            Assert.False(attribute.ConvertEmptyStringToNull);
+        }
+
+        [Fact]
+        public void ApplyFormatInEditMode_Get_Set()
+        {
+            DisplayFormatAttribute attribute = new DisplayFormatAttribute();
+            attribute.ApplyFormatInEditMode = true;
+
+            Assert.True(attribute.ApplyFormatInEditMode);
+        }
+
+        [Fact]
+        public void HtmlEncode_Get_Set()
+        {
+            DisplayFormatAttribute attribute = new DisplayFormatAttribute();
+            attribute.HtmlEncode = false;
+
+            Assert.False(attribute.HtmlEncode);
+        }
+
+        public static IEnumerable<object[]> Strings_TestData()
+        {
+            yield return new object[] { "" };
+            yield return new object[] { " \r \t \n " };
+            yield return new object[] { "abc" };
+        }
+
+        [Theory]
+        [MemberData(nameof(Strings_TestData))]
+        [InlineData("NullDisplayText")]
+        public void NullDisplayText_Get_Set(string input)
+        {
+            DisplayFormatAttribute attribute = new DisplayFormatAttribute();
+            attribute.NullDisplayText = input;
+
+            Assert.Equal(input, attribute.NullDisplayText);
+            Assert.Equal(input == null, attribute.GetNullDisplayText() == null);
```

Comment:
`Assert.NotNull(attribute.GetNullDisplayText())` since `input` is never `null`. Note: Test `null` / untouched case in the _separate_ test I mentioned above.